File: eParking.Services/Database/Parking/FavoriteParkingLot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eParking.Services.Database.Parking
{
    public class FavoriteParkingLot
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public MyAppUser User { get; set; } = null!;

        [Required]
        public int ParkingLotId { get; set; }

        [ForeignKey(nameof(ParkingLotId))]
        public ParkingLot ParkingLot { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: UI/eParking_mobile/lib/models/favorite_parking_lot.dart
class FavoriteParkingLot {
  final int id;
  final int userId;
  final int parkingLotId;
  final String parkingLotName;
  final DateTime createdAt;

  FavoriteParkingLot({
    required this.id,
    required this.userId,
    required this.parkingLotId,
    required this.parkingLotName,
    required this.createdAt,
  });

  factory FavoriteParkingLot.fromJson(Map<String, dynamic> json) {
    return FavoriteParkingLot(
      id: json['id'] as int,
      userId: json['userId'] as int,
      parkingLotId: json['parkingLotId'] as int,
      parkingLotName: json['parkingLotName']?.toString() ?? '',
      createdAt: DateTime.parse(json['createdAt'].toString()),
    );
  }
}
